sdks/java/core/src/main/java/org/apache/beam/sdk/io/FileSystems.java

```diff
@@ -108,7 +94,8 @@ static PipelineOptions getDefaultConfig(String scheme) {
   static FileSystem getFileSystemInternal(URI uri) {
     String lowerCaseScheme = (uri.getScheme() != null
         ? uri.getScheme().toLowerCase() : LocalFileSystemRegistrar.LOCAL_FILE_SCHEME);
-    return getRegistrarInternal(lowerCaseScheme).fromOptions(getDefaultConfig(lowerCaseScheme));
+    return getRegistrarInternal(lowerCaseScheme)
+        .fromOptions(checkNotNull(defaultConfig, "defaultConfig"));
```

Comment:
Done

Didn't include the URI, since the check fail will happen for any scheme for now.